service/matching/taskQueueManager.go

```diff
@@ -343,9 +331,7 @@ func (c *taskQueueManagerImpl) WaitUntilInitialized(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	// We don't really care if the initial fetch worked or not, anything that *requires* a bit of metadata should fail
-	// that operation if it's never fetched OK. If the initial fetch errored, the metadataPoller will have been started.
-	_, _ = c.userDataInitialFetch.Get(ctx)
+	_, err = c.userDataInitialFetch.Get(ctx)
```

Comment:
I'm wondering if we should be waiting for the initial fetch here at all, AFAICT, we'll be serving concurrent requests on this versioned task queue partition, each with separate deadlines.
Shouldn't we block indefinitely here?
What are the cases where we'd want to return this error to the caller?

Reply:
This is called by rpc handlers. We can't block indefinitely, we need to return an error so the rpc can complete (with an error)